### ISSUE TITLE
Does not work with eager mode

### ISSUE THREAD

**Issue by qinghao1** [tensorflow/privacy#87]
To reproduce:

1. Go to `tutorials/Classification_Privacy.ipynb`
2. Run the following
```
tf.compat.v1.enable_eager_execution()
assert tf.executing_eagerly()
```
3. Run the notebook. Accuracy does not increase with training, stays at 0.1. If I remove the above two lines and rerun, accuracy increases as expected.

**Comment by skeydan** [comment]
I can confirm on my own data. Running TF 2 or TF nightly, whether using subclassing or Keras, the losses and weights are all `nan`.
I have to `disable_v2_behavior` to make it work.

It would be great if this could be fixed, thanks!

**Comment by carlodavid012** [comment]
what's the update to this? does tf-privacy still doesn't work on eager mode?
